helper/linker.go

```
@@ -106,52 +112,102 @@ func (f FileLinker) Configure(appDir string) error {
 		if err := server.SetUserDirectory(usrPath, destUserPath, serverName); err != nil {
 			return fmt.Errorf("unable to contribute packaged server\n%w", err)
 		}
-	} else {
-		if err = f.ContributeApp(appDir, f.RuntimeRootPath, serverName, b); err != nil {
-			return fmt.Errorf("unable to contribute app and config to runtime root\n%w", err)
-		}
+		return nil
+	}
 
-		if err = f.ContributeDefaultHttpEndpoint(f.RuntimeRootPath, serverName, b); err != nil {
-			return fmt.Errorf("unable to contribute default http endpoint\n%w", err)
-		}
+	// Contribute server config files if found in workspace
+	configs := []string{
+		"server.xml",
+		"server.env",
+		"bootstrap.properties",
+		"jvm.options",
+	}
 
-		if err = f.ContributeUserFeatures(serverName, f.getConfigTemplate(b, "features.tmpl")); err != nil {
-			return fmt.Errorf("unable to contribute user features\n%w", err)
+	for _, config := range configs {
+		configPath := filepath.Join(workspacePath, config)
+		configExists, err := util.FileExists(configPath)
+		if err != nil {
+			return err
+		}
+		if configExists {
+			toPath := filepath.Join(f.ServerRootPath, config)
```

I think the way you'd typically handle this in Go is to just try the file operation, so just run `util.DeleteAndLinkPath(..)` but then check the return `err` to see if it matches a file not found error (see `os.IsNotExist(err error)`. If it's file not found, you ignore the error and move on.

Doing it this way avoids potential race conditions between checking if the file exists and actually performing the file operation. I'm not sure that race conditions matters a ton here, but I do think doing it this way would be more idiomatic Go code.